web/packages/teleterm/src/services/appUpdater/autoUpdatesStatus.ts

```diff
@@ -127,6 +138,24 @@ function findVersionFromClusters(sources: {
   };
 }
 
+/** When `false` is returned, the user will need to click 'Download' manually. */
+export function shouldAutoDownload(updatesStatus: AutoUpdatesEnabled): boolean {
+  if (updatesStatus.source.kind !== 'most-compatible') {
+    return true;
+  }
```

Comment:
Nit: I feel like it'd be more readable to spell out the choices with a switch instead. Otherwise when reading this function you have to remember that: 1) there are three different kinds available 2) the rest of the function body considers only a situation where `updatesStatus.source.kind === 'most-compatible'`. A switch can present complex cases in a clearer form, I think.